core/eolearn/tests/test_utils/test_parsing.py

```diff
@@ -0,0 +1,142 @@
+from dataclasses import dataclass
+from typing import List, Tuple, Union
+
+import numpy as np
+import pytest
+
+from eolearn.core import EOPatch, FeatureParser, FeatureType
+from eolearn.core.utils.parsing import FeatureRenameSpec, FeatureSpec, FeaturesSpecification
+from eolearn.core.utils.types import EllipsisType
+
+
+@dataclass
+class ParsingTestClass:
+    input: FeaturesSpecification
+    output_get_features: List[FeatureSpec]
+    output_get_renamed_features: List[FeatureRenameSpec]
+    output_get_feature_specifications: List[Tuple[FeatureType, Union[str, EllipsisType]]]
+
+
+TEST_CASES_EMPTY = [
+    ParsingTestClass(
+        input=(), output_get_features=[], output_get_renamed_features=[], output_get_feature_specifications=[]
+    ),
+]
+
+
+TEST_CASES = [
```

Comment:
missing a single feature case, also often include FeatureType.BBOX and TIMESTAMP as they have different behavior

Reply:
done